### PR TITLE
CDAP-19230: convert pipeline canvas actions tests

### PR DESCRIPTION
**Pull request by de-lan**
# Convert pipeline canvas actions tests

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [x] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-19230](https://cdap.atlassian.net/browse/CDAP-19230)

## Test Plan
Manual
## Screenshots
image

### CODE REVIEW

File: src/e2e-test/features/pipeline.canvas.actions.feature [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# License for the specific language governing permissions and limitations under
+# the License.
+#
+
+@Integration_Tests
+Feature: Pipeline Canvas - Actions
+
+  @PIPELINE_CANVAS_ACTIONS_TEST
+  Scenario: Correctly undo redo actions done by the user
+    When Open pipeline studio page
```
Comment (sumengwang) [inline src/e2e-test/features/pipeline.canvas.actions.feature:22]: You have `Given` tag for the function but here you have `When`

**Review by sumengwang — APPROVED** [review]

File: src/e2e-test/features/pipeline.canvas.actions.feature (hunk 1 above)
Comment (de-lan) [reply src/e2e-test/features/pipeline.canvas.actions.feature:22]: nice catch!

File: src/e2e-test/java/io/cdap/cdap/ui/utils/Commands.java [hunk 2]
```diff
@@ -31,35 +43,183 @@ public static void addNodeToCanvas(NodeInfo node) {
...
+    connectTwoNodes(joinerNode, conditionNode);
+
+    connectTwoNodes(conditionNode, sinkNode1);
+    connectTwoNodes(conditionNode, sinkNode2);
+
+    pipelineCleanUpGraphControl();
+    fitPipelineToScreen();
+  }
+
+  public static void takeAScreenshot() {
```
Comment (de-lan) [inline src/e2e-test/java/io/cdap/cdap/ui/utils/Commands.java:314]: This function is more of an assistant purpose, which helps to have a viewable screenshot at a testing step

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineCanvasActions.java [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+  @When("Open pipeline studio page")
+  public void openPipelineStudioPage() {
+    try {
+      SeleniumDriver.openPage(Constants.PIPELINE_STUDIO_URL);
+      WaitHelper.waitForPageToLoad();
+    } catch (UnhandledAlertException e) {
+      try {
+        Alert alert = SeleniumDriver.getDriver().switchTo().alert();
+        String alertText = alert.getText();
+        System.out.println("Alert data: " + alertText);
```
Comment (radhikav1) [inline src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineCanvasActions.java]: system.out can be removed

File: src/e2e-test/java/io/cdap/cdap/ui/utils/Commands.java [hunk 4]
```diff
@@ -31,35 +43,183 @@ public static void addNodeToCanvas(NodeInfo node) {
...
   public static void fitPipelineToScreen() {
     ElementHelper.clickOnElement(
       Helper.locateElementByCssSelector("[data-testid=\"pipeline-fit-to-screen-control\"]")
     );
-  };
+  }
+
+  public static void toggleTransformPanel() {
+    ElementHelper.clickOnElement(
+      Helper.locateElementByCssSelector("[data-testid=\"plugin-Transform-group-summary\"]")
```
Comment (radhikav1) [inline src/e2e-test/java/io/cdap/cdap/ui/utils/Commands.java]: Could use getCssSelectorByDataTestId(..)

File: src/e2e-test/java/io/cdap/cdap/ui/utils/Commands.java [hunk 5]
```diff
@@ -31,35 +43,183 @@ public static void addNodeToCanvas(NodeInfo node) {
...
+    );
+  }
+
+  public static void toggleAnalyticsPanel() {
+    ElementHelper.clickOnElement(
+      Helper.locateElementByCssSelector("[data-testid=\"plugin-Analytics-group-summary\"]")
```
Comment (radhikav1) [inline src/e2e-test/java/io/cdap/cdap/ui/utils/Commands.java]: Could use getCssSelectorByDataTestId(..) wherever applicable

Comment (de-lan) [reply src/e2e-test/java/io/cdap/cdap/ui/utils/Commands.java]: Thanks! Updated these selectors

**Review by radhikav1 — APPROVED** [review]

**Pull request merged**